docs/content/integrations/openai.mdx

```diff
@@ -0,0 +1,140 @@
+
+---
+
+## Getting started
+
+Before you get started with the `dagster-openai`library, we recommend familiarizing yourself with the [OpenAI Python API library](https://github.com/openai/openai-python), which this integration uses to interact with the [OpenAI REST API](https://platform.openai.com/docs/introduction).
+
+---
+
+## Prerequisites
+
+To get started, install the `dagster` and `dagster-openai` Python packages:
+
+```bash
+pip install dagster dagster-openai
+```
+
+Note that you will need an OpenAI [API key](https://platform.openai.com/api-keys) to use the resource, which can be generated in your OpenAI account.
+
+---
+
+## Connecting to OpenAI
+
+The first step in using OpenAI with Dagster is to tell Dagster how to connect to an OpenAI client using an OpenAI [resource](/concepts/resources). This resource contains the credentials needed to interact with OpenAI API.
+
+We will supply our credentials as environment variables by adding them to a `.env` file. For more information on setting environment variables in a production setting, see [Using environment variables and secrets](/guides/dagster/using-environment-variables-and-secrets).
+
+```bash
+# .env
+
+OPENAI_API_KEY=...
+```
+
+Then, we can instruct Dagster to authorize the OpenAI resource using the environment variables:
+
+```python startafter=start_example endbefore=end_example file=/integrations/openai/resource.py dedent=4
+dagster_openai import OpenAIResource
+
+ dagster import EnvVar
+
+ll API key from environment variables
+ai = OpenAIResource(
+api_key=EnvVar("OPENAI_API_KEY"),
+```
+
+---
+
+## Using the OpenAI resource with assets
+
+The OpenAI resource can be used in assets in order to interact with the OpenAI API. Note that in this example, we supply our credentials as environment variables directly when instantiating the <PyObject object="Definitions" /> object.
+
+```python startafter=start_example endbefore=end_example file=/integrations/openai/assets.py dedent=4
+dagster_openai import OpenAIResource
+
+ dagster import (
+AssetExecutionContext,
+Definitions,
+EnvVar,
+asset,
+define_asset_job,
+
+
+
+et(compute_kind="OpenAI")
+openai_asset(context: AssetExecutionContext, openai: OpenAIResource):
+with openai.get_client(context) as client:
+    client.chat.completions.create(
+        model="gpt-3.5-turbo",
+        messages=[{"role": "user", "content": "Say this is a test."}],
+    )
+
+
+ai_asset_job = define_asset_job(name="openai_asset_job", selection="openai_asset")
+
+ = Definitions(
+assets=[openai_asset],
+jobs=[openai_asset_job],
+resources={
+    "openai": OpenAIResource(api_key=EnvVar("OPENAI_API_KEY")),
+},
+```
+
+After materializing your asset, your OpenAI API usage metadata will be available in the **Events** and **Plots** tabs of your asset in the Dagster UI. Refer to the [Viewing and materializing assets in the UI guide](https://docs.dagster.io/concepts/assets/software-defined-assets#viewing-and-materializing-assets-in-the-ui) for more information.
```

Comment:
should we navigate to Insights here? something like - if Dagster Cloud, the plots could show up in Insights, blah blah.

Reply:
➕